### PR TITLE
feat: [CHK-2982] Added tracing for queue messages

### PR DESCRIPTION
**Pull request by petretiandrea**
Storage queue tracing using span link

#### List of Changes
- instrumentation through Spring Integration Otel, instead of Spring scheduling
- added span link to trace which generate the event
- fix missing span children. Now every operation inside the consumer is showed down to the consumer span
- bump of spring boot and azure deps to fix an issue with storage queue span

#### How Has This Been Tested?
Tested in DEV
Screenshot 2024-06-27 alle 11 35 04
Screenshot 2024-06-27 alle 11 36 07
Screenshot 2024-06-27 alle 11 36 33

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by giovanniberti — APPROVED** [review]
Good job! 🚀

**Review by pietro-tota — DISMISSED** [review]
Well done 🚀  just an observation :D

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -3,31 +3,27 @@
    <configuration>
       <verify-metadata>true</verify-metadata>
       <verify-signatures>false</verify-signatures>
-      <trusted-artifacts>
-         <trust file=".*-javadoc[.]jar" regex="true"/>
-         <trust file=".*-sources[.]jar" regex="true"/>
-      </trusted-artifacts>
```
Comment (pietro-tota) [inline gradle/verification-metadata.xml:9]: why removing those artifacts? they are needed for local project build

**Comment by pietro-tota** [comment]
please check CR, it's failing for missing artifact verification

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (petretiandrea) [reply gradle/verification-metadata.xml]: fixed by https://github.com/pagopa/pagopa-payment-wallet-event-dispatcher-service/pull/26/commits/7599761a167d50c40ed27472c2b37661c58979ba

**Review by giovanniberti — APPROVED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**